x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -480,9 +480,14 @@ public static DataType getTargetType(String[] names) {
             return UNSUPPORTED;
         }
 
-        return types.stream()
-            .min((dt1, dt2) -> dataTypeCastingPriority.get(dt1).compareTo(dataTypeCastingPriority.get(dt2)))
-            .orElse(UNSUPPORTED);
+        try {
+            return types.stream()
+                .min((dt1, dt2) -> dataTypeCastingPriority.get(dt1).compareTo(dataTypeCastingPriority.get(dt2)))
```

Comment:
I spent a lot of time digging into why tests were failing here.  Ultimately, it's unclear to me why the other functions supporting date nanos were not throwing an NPE here.  The NPE happens because `dataTypeCastingPriority.get(DATE_NANOS)` returns `null`, since date nanos are not in that map, which then causes `compareTo` to fail with the observed NPE.

My understanding is that `dataTypeCastingPriority` is used for autocasting literals, which is not currently planned for date nanos(note that autocasting literals is different from the planned auto-union-type work in https://github.com/elastic/elasticsearch/issues/110009).  Given that, it seemed like adding `DATE_NANOS` to the map would result in incorrect behavior.  Having said that, it is deeply unclar to me if catching the exception and returning `UNSUPPORTED` is correct here.  Advice would be welcome.